frontend/packages/dev-console/src/components/monitoring/alerts/MonitoringAlerts.tsx

```diff
@@ -0,0 +1,154 @@
+import * as React from 'react';
+import Helmet from 'react-helmet';
+import * as _ from 'lodash';
+// FIXME upgrading redux types is causing many errors at this time
+// eslint-disable-next-line @typescript-eslint/ban-ts-ignore
+// @ts-ignore
+import { useDispatch, connect } from 'react-redux';
+import { match as RMatch } from 'react-router-dom';
+import { Table, TableHeader, TableBody, SortByDirection } from '@patternfly/react-table';
+import { FilterToolbar } from '@console/internal/components/filter-toolbar';
+import { getAlertsAndRules } from '@console/internal/components/monitoring/utils';
+import { monitoringSetRules } from '@console/internal/actions/ui';
+import { useURLPoll } from '@console/internal/components/utils/url-poll-hook';
+import { PrometheusRulesResponse, Rules } from '@console/internal/components/monitoring/types';
+import { PROMETHEUS_TENANCY_BASE_PATH } from '@console/internal/components/graphs';
+import { RootState } from '@console/internal/redux';
+import {
+  alertSeverityOrder,
+  alertStateOrder,
+  alertingRuleStateOrder,
+} from '@console/internal/reducers/monitoring';
+import { getFilteredRows } from '@console/internal/components/factory';
+import {
+  monitoringAlertRows,
+  monitoringAlertColumn,
+  alertFilters,
+} from './monitoring-alerts-utils';
+import './MonitoringAlerts.scss';
+
+type MonitoringAlertsProps = {
+  match: RMatch<{
+    ns?: string;
+  }>;
+};
+
+type StateProps = {
+  rules: Rules;
+  filters: { [key: string]: any };
+};
+
+type props = MonitoringAlertsProps & StateProps;
+
+const POLL_DELAY = 15 * 1000;
+const reduxID = 'devMonitoringAlerts';
+const textFilter = 'alert-list-text';
```

Comment:
`alert-list-text` is specifically for alerts, not rules. You can either use `alerting-rule-name`, which just matches against the rule name, or you can create a new function like `alerting-rule-list-text` based on `alert-list-text`.

Reply:
rename `alert-list-text` as `resource-list-text` so that we can use it for both alerts and rules.